### PR TITLE
[CodeGen] Add ValueType v3i8 (NFCI).

### PR DESCRIPTION
**Pull request by fpetrogalli**

### CODE REVIEW

**Comment by fpetrogalli** [comment]
Hang on, genuine failures in https://buildkite.com/llvm-project/github-pull-requests/builds/36346#018d7e21-6650-4dd6-b687-9f472dbd9145


```
  LLVM :: CodeGen/AArch64/vec3-loads-ext-trunc-stores.ll
```

**Review by francisvm — APPROVED** [review]

**Comment by fpetrogalli** [comment]
The current version will fail these two AMDGPU CodeGen tests: 

```
  LLVM :: CodeGen/AMDGPU/calling-conventions.ll
  LLVM :: CodeGen/AMDGPU/store-v3i64.ll
```

@arsenm, are you able to look into this? I blamed places in the AMDGPU backend that mention truncating stores to i8, you seem to have workedd with them. Can you help out here? My knowledge of this ISA is close to zero...

**Comment by arsenm** [comment]
You probably need to update the setTruncStoreAction/setLoadExtActions, the current set go through the existing i8 vectors

**Comment by fpetrogalli** [comment]
> You probably need to update the setTruncStoreAction/setLoadExtActions, the current set go through the existing i8 vectors

Thank you @arsenm , that did it locally. Let's see what buildkite has to say.

Francesco

File: llvm/lib/Target/AMDGPU/AMDGPUISelLowering.cpp [hunk 1]
```diff
@@ -308,8 +308,11 @@ AMDGPUTargetLowering::AMDGPUTargetLowering(const TargetMachine &TM,
   setTruncStoreAction(MVT::v2f64, MVT::v2f32, Expand);
   setTruncStoreAction(MVT::v2f64, MVT::v2f16, Expand);
 
+  setTruncStoreAction(MVT::v3i32, MVT::v3i8, Expand);
```
Comment (arsenm) [inline llvm/lib/Target/AMDGPU/AMDGPUISelLowering.cpp:311]: I suspect we're missing test coverage (or we're doing unnecessary work) for the load half of this

**Review by arsenm — APPROVED** [review]

File: llvm/lib/Target/AMDGPU/AMDGPUISelLowering.cpp (hunk 1 above)
Comment (fpetrogalli) [reply llvm/lib/Target/AMDGPU/AMDGPUISelLowering.cpp:311]: @arsenm - may I merge the patch without taking care of this?

Comment (arsenm) [reply llvm/lib/Target/AMDGPU/AMDGPUISelLowering.cpp:311]: yes

**Pull request merged**